### PR TITLE
[next] FS create folder before writing in it

### PR DESCRIPTION
**Pull request by WebReflection**
## Description

See https://github.com/pyscript/pyscript/issues/1580

## Changes

  * apply suggested changes to avoid analyzePath to throw
  * drop irrelevant unit tests around it (requires integration)

## Checklist

-   [x] All tests pass locally
-   [ ] I have updated `docs/changelog.md`
-   [ ] I have created documentation for this(if applicable)

### CODE REVIEW

**Comment by JeffersGlass** [comment]
Suspect this wants to target `next` and not `main` yes? Also it looks like #1581 was submitted, which is very similar.

**Comment by WebReflection** [comment]
@JeffersGlass yes, the default to *master* instead of *next* is making my days lovely ... but also this one, as MR, doesn't ruin code coverage so I think this should be considered for now.

**Comment by e-nikolov** [comment]
Looks like next to its FS API, Emscripten also has [PATH and PATH_FS](https://github.com/emscripten-core/emscripten/blob/main/src/library_path.js) that include things like dirname(), basename() and resolve(). Would it make sense to use them?

File: pyscript.core/esm/interpreter/_utils.js [hunk 1]
```diff
@@ -40,11 +39,12 @@ export const writeFile = (FS, path, buffer) => {
         canOwn: true,
     });
 };
+/* c8 ignore stop */
 
 // This is instead a fallback for Lua or others
 export const writeFileShim = (FS, path, buffer) => {
-    path = resolve(FS, path);
     mkdirTree(FS, dirname(path));
```
[the remark this answers is not in the document]
Comment (hoodmane) [reply pyscript.core/esm/interpreter/_utils.js:46]: > why pyodide doesn't just allow access to `FS_PATH`

Haven't gotten around to it. I'm thinking about merging `FS_PATH` and `PATH` and exposing them as `pyodide.PATH` or `pyodide.FS.path` or something like that. But in the meantime, `pyodide._module.FS_PATH` is not likely to go anywhere.

File: pyscript.core/esm/interpreter/_utils.js [hunk 2]
```diff
@@ -30,7 +30,6 @@ export const stdio = (init) => {
         },
     };
 };
-/* c8 ignore stop */
 
 // This should be the only helper needed for all Emscripten based FS exports
 export const writeFile = (FS, path, buffer) => {
```
[the remark this answers is not in the document]
Comment (WebReflection) [reply pyscript.core/esm/interpreter/_utils.js]: Done

File: pyscript.core/esm/interpreter/_utils.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (WebReflection) [reply pyscript.core/esm/interpreter/_utils.js:46]: For now I am using `._module` as both Pyodide and MicroPython have some hidden stuff in it (in different places) but I am currently happy about the state of this MR.

**Comment by WebReflection** [comment]
@e-nikolov I think I have addressed all concerns and now it's fully based on native functionality.

One notable difference between pyodide and MicroPython is that Pyodide consider its root the `/pyodide/` folder while MicroPython doesn't have a special root folder.

This is, however, aligned with what previous PyScript had too so I think we should be good.

All manual tests passed without issues.

**Comment by WebReflection** [comment]
Added an issue to *wasmoon* project: https://github.com/ceifa/wasmoon/issues/83

I'd love to fully get rid of the current ugly, quick, and inevitably dirty, code around that interpreter too.

**Review by tedpatrick — APPROVED** [review]

**Review by FabioRosado — APPROVED** [review]

**Pull request merged**